spec/models/concerns/digital_object_concerns/publish_behavior_spec.rb

```diff
@@ -4,48 +4,58 @@
 
 RSpec.describe DigitalObjectConcerns::PublishBehavior do
   include_context 'with stubbed search adapters'
-  let(:publishing_user) { FactoryBot.create(:user) }
-  let(:project) { FactoryBot.create(:project) }
-  let(:publish_target_1) do
-    FactoryBot.create(:publish_target)
-  end
-  let(:publish_target_2) do
-    FactoryBot.create(:publish_target)
-  end
-
-  let(:digital_object_without_publish_entries) { FactoryBot.create(:digital_object_test_subclass) }
-  let(:digital_object_with_publish_entries) do
-    obj = FactoryBot.create(:digital_object_test_subclass)
-    obj.send(
-      :publish_entries=,
-      [
-        PublishEntry.new(publish_target: publish_target_1, published_at: Time.current, published_by: publishing_user),
-        PublishEntry.new(publish_target: publish_target_2, published_at: Time.current, published_by: publishing_user)
-      ]
-    )
-    obj
-  end
 
   describe "#perform_publish_changes" do
+    let(:publish_target) { FactoryBot.create(:publish_target, doi_priority: 1, is_allowed_doi_target: true) }
+    let(:object) { FactoryBot.create(:digital_object_test_subclass, doi: doi) }
+    let(:doi) { '10.abcd/4569' }
+    let(:target_url) { "http://example.org/objects/#{object.uid}" }
+    let(:publish_entry) { PublishEntry.new(publish_target: publish_target, citation_location: target_url) }
+    let(:publish_promise) do
+      Concurrent::Promise.new { publish_entry }.execute
+    end
     context 'to a publish target that is an allowed DOI target' do
-      let(:publish_target) { FactoryBot.create(:publish_target, doi_priority: 1, is_allowed_doi_target: true) }
-      let(:object) { FactoryBot.create(:digital_object_test_subclass, doi: doi) }
-      let(:doi) { '10.abcd/4569' }
-      let(:location_uri) { "http://example.org/objects/#{object.uid}" }
-      let(:publish_entry) { PublishEntry.new(publish_target: publish_target, citation_location: location_uri) }
-      let(:publish_promise) do
-        Concurrent::Promise.new { publish_entry }.execute
+      before do
+        # pretend we were preserved 10 seconds ago
+        object.preserved_at = DateTime.new.getlocal - 10
       end
-
       it "calls #update on external identifier adapter" do
         expect(object).to receive(:async_publish).and_return(publish_promise)
-        expect(Hyacinth::Config.external_identifier_adapter).to receive(:update).with(doi, digital_object: object, location_uri: location_uri)
+        expect(Hyacinth::Config.external_identifier_adapter).to receive(:update).with(doi, digital_object: object, target_url: target_url)
         object.perform_publish_changes(publish_to: [publish_target])
       end
     end
+    context 'when an object has not been preserved' do
+      it "raises an exception without starting asynch processes" do
```

Comment:
```suggestion
      it "raises an exception without starting async processes" do
```
This is purely personal preference, so feel free to ignore this suggestion!

I usually use "async" rather than "asynch" as shorthand for "asynchronous", and this has been especially reinforced for me by the `async` function keyword in JavaScript.